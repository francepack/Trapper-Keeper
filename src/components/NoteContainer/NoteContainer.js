import React from 'react';
import Note from '../../containers/Note/Note'

export const NoteContainer = (props) => {
  console.log(props.notes)
  const notes = props.notes.map(note => <Note key={note.id} id={note.id} title={note.title} items={note.items} />)
  return (
    <div>
      {notes}
    </div>
  )
}

export default NoteContainer