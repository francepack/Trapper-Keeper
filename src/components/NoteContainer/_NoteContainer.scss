@import '../../helpers/variables';

.note-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.view-note {
  width: 25%;
  height: 100%;
  margin: 25px;
  min-width: 325px;
    &:hover {
      cursor: pointer;
      border: solid $logo 1px;
      transition: border .5s ease;
    }
}

.view-item {
  &:hover {
    cursor: pointer;
  }
}