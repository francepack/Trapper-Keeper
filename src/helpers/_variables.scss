
$background: #fff;
$logo: #f4b70b;
$placeholder: #767676;
$type: #202124;
$icon: #606163;
$hover: #e8eaed;
$form-border: #e0e0e0;
$save-btn: #4285F4;
$save-btn-hover: #4285f4d6;
$check-mark: #0F9D58;
$delete-btn: #DB4437;
$delete-hover: #db4437c9;