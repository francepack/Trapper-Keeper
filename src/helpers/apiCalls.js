export const addNote = async (note) => {
  const url = 'http://localhost:3001/api/v1/notes'
  try {
    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify(note),
      headers: {
        'Content-Type': 'application/json'
      },
    })
    const addedNote = await response.json()
    return addedNote;
  } catch (error) {
    return error.message
  }
}

export const getNotes = async () => {
  const url = 'http://localhost:3001/api/v1/notes'
  try {
    const response = await fetch(url);
    const allNotes = await response.json();
    return allNotes.notes
  } catch (error) {
    return error.message
  }
}

export const removeNote = async (id) => {
  const url = `http://localhost:3001/api/v1/notes/${id}`
  try {
    const response = await fetch(url, {
      method: 'DELETE',
      body: JSON.stringify(id),
      headers: {
        'Content-Type': 'application/json'
      },
    })

  } catch(error) {
    return error.message
  }
}