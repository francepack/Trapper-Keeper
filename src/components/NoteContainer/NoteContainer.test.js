import React from 'react'
import { NoteContainer } from './NoteContainer'
import { mockAllNotes } from '../../helpers/mockData'
import { shallow } from 'enzyme'

describe('NoteContainer', () => {
  let wrapper

  it('should match snapshot', () => {
    wrapper = shallow(
      <NoteContainer {...mockAllNotes} />
    )

      expect(wrapper).toMatchSnapshot()
  })
})
