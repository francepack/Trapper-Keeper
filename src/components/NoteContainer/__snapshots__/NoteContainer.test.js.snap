// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteContainer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteContainer
    notes={
      Array [
        Object {
          "id": 4,
          "items": Array [
            "item4a",
            "item4b",
          ],
          "title": "Title Four",
        },
        Object {
          "id": 5,
          "items": Array [
            "item5a",
            "item5b",
          ],
          "title": "Title Five",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ViewNote
          note={
            Object {
              "id": 4,
              "items": Array [
                "item4a",
                "item4b",
              ],
              "title": "Title Four",
            }
          }
        />,
        <ViewNote
          note={
            Object {
              "id": 5,
              "items": Array [
                "item5a",
                "item5b",
              ],
              "title": "Title Five",
            }
          }
        />,
      ],
      "className": "note-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "note": Object {
            "id": 4,
            "items": Array [
              "item4a",
              "item4b",
            ],
            "title": "Title Four",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "note": Object {
            "id": 5,
            "items": Array [
              "item5a",
              "item5b",
            ],
            "title": "Title Five",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ViewNote
            note={
              Object {
                "id": 4,
                "items": Array [
                  "item4a",
                  "item4b",
                ],
                "title": "Title Four",
              }
            }
          />,
          <ViewNote
            note={
              Object {
                "id": 5,
                "items": Array [
                  "item5a",
                  "item5b",
                ],
                "title": "Title Five",
              }
            }
          />,
        ],
        "className": "note-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "note": Object {
              "id": 4,
              "items": Array [
                "item4a",
                "item4b",
              ],
              "title": "Title Four",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "note": Object {
              "id": 5,
              "items": Array [
                "item5a",
                "item5b",
              ],
              "title": "Title Five",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
